src/custom_component_renderer/book_directory_renderer.rs

```diff
@@ -0,0 +1,264 @@
+        root: PathBuf,
+    ) -> Self {
+        RendererConfiguration {
+            i18n,
+            destination,
+            current_language,
+            root,
+        }
+    }
+}
+
+pub struct RenderingContext {
+    pub path: PathBuf,
+    pub destination: PathBuf,
+    pub language: String,
+}
+
+impl RenderingContext {
+    fn new(path: PathBuf, destination: PathBuf, language: String) -> Self {
+        RenderingContext {
+            path,
+            destination,
+            language,
+        }
+    }
+
+    pub fn rendered_path(&self) -> String {
+        let mut relative_path = PathBuf::from(
+            self.path
+                .strip_prefix(&self.destination)
+                .expect("Invalid path"),
+        );
+        if let Ok(stripped) = relative_path.strip_prefix(&self.language) {
+            relative_path = stripped.to_owned();
+        }
+        String::from(
+            relative_path
+                .to_str()
+                .expect("Failed to convert path to rendered path"),
+        )
+    }
+}
+
+pub(crate) struct BookDirectoryRenderer {
+    config: RendererConfiguration,
+    components: Vec<Box<dyn Component>>,
+    languages_paths: BTreeMap<String, PathBuf>,
+}
+
+impl BookDirectoryRenderer {
+    pub(crate) fn new(config: RendererConfiguration) -> BookDirectoryRenderer {
+        let default_language = config.i18n.default_language.clone();
+        let languages_paths = config
+            .i18n
+            .languages
+            .keys()
+            .filter(|language| {
+                default_language.is_none() || *language != default_language.as_ref().unwrap()
+            })
+            .map(|language| {
+                (
+                    language.clone(),
+                    config.destination.join("html").join(language),
+                )
+            })
+            .collect::<BTreeMap<String, PathBuf>>();
+        BookDirectoryRenderer {
+            config,
+            languages_paths,
+            components: Vec::new(),
+        }
+    }
+
+    pub fn translate(&self) -> Result<()> {
+        let default_language = &self.config.i18n.default_language;
+        for (identifier, _) in &self.config.i18n.languages {
+            if let Some(default_language) = default_language {
+                if default_language == identifier {
+                    continue;
+                }
+            }
+
+            let destination = self.config.destination.as_path().display().to_string();
+            let book_folder = self.config.root.as_path().display().to_string();
+
+            Command::new("mdbook")
+                .arg("build")
+                .arg(&book_folder)
+                .arg("-d")
+                .arg(&format!("{destination}/{identifier}"))
+                .env("MDBOOK_BOOK__LANGUAGE", identifier)
+                .env(
+                    "MDBOOK_OUTPUT__HTML__SITE_URL",
+                    &format!("/comprehensive-rust/{identifier}/"),
```

Comment:
This looks hard-coded to work with Comprehensive Rust :slightly_smiling_face: The tooling here is used by 5+ other projects right now (see the README), so it has to be generic.

Reply:
Yes, this part is not ready yet. I will remove the need to spawn new processes by directly calling `book.clone()` -> `get_text(book)` ->  `book.build()` but I first need to move the methods in `get_text` out into their own library

Reply:
Also now, with the `mdbook build` process we don't get hot reloading because it's not added by default